src/core/Provider.ts

```diff
@@ -1,51 +1,87 @@
-import React, { createElement, useRef, useDebugValue } from 'react'
+import React, { createElement, useCallback, useRef, useDebugValue } from 'react'
 
 import type { AnyAtom, Scope } from './types'
+import { subscribeAtom } from './vanilla'
 import type { AtomState, State } from './vanilla'
 import { createStore, getStoreContext } from './contexts'
+import type { Store } from './contexts'
 import { useMutableSource } from './useMutableSource'
 
-const getState = (state: State) => state
-// TODO this doesn't trigger re-render so useDebugState can be stale.
-const subscribe = (_state: State, _callback: () => void) => () => {}
-
 export const Provider: React.FC<{
   initialValues?: Iterable<readonly [AnyAtom, unknown]>
   scope?: Scope
 }> = ({ initialValues, scope, children }) => {
   const storeRef = useRef<ReturnType<typeof createStore> | null>(null)
-  if (storeRef.current == null) {
-    // lazy initialization
-    storeRef.current = createStore(initialValues)
-  }
-  const store = storeRef.current as ReturnType<typeof createStore>
 
-  if (typeof process === 'object' && process.env.NODE_ENV !== 'production') {
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    useDebugState(useMutableSource(store[0], getState, subscribe))
+  if (
+    typeof process === 'object' &&
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test'
```

Comment:
I do see the assertion on `process`, but could this potentially screw with consumers test builds?

Reply:
This is newly added in this PR. Wish there would be a way to avoid this.
Without it, the test fails because the provider invokes extra subscribes.